### ISSUE TITLE
VPN provider support: AdguardVPN

### ISSUE THREAD

**Issue by alx-xlx** [qdm12/gluetun#2076]
The list of servers are hidden behind adguardvpn website. It has recently been released for linux usage.

Here is the link https://www.reddit.com/r/Adguard/comments/182s3f7/adguard_vpn_now_supports_routers/

and this is how the configuration items look like. taken from one of the comments suggesting how to use it on linux

![](https://i.postimg.cc/wMLDcYkQ/adgipsec.png)

**Comment by leo15dev** [comment]
[AdGuard VPN now supports routers](https://adguard-vpn.com/en/blog/adguard-vpn-for-routers.html)

This feature requires an [AdGuard VPN subscription](https://adguard-vpn.com/license.html) and IKEv2/IPsec support. More protocols will be added in the future.

AdGuard VPN cannot be used on Gluetun unless it supports WireGuard or OpenVPN. So it's been a long wait!

**Comment by MrKaon** [comment]
The long wait isn't over yet.

**Comment by britshickhouse** [comment]
Yes please, +1 for AdguardVPN support. There's this page about the protocol in their docs which might be useful: [https://adguard-vpn.com/kb/general/adguard-vpn-protocol/](https://adguard-vpn.com/kb/general/adguard-vpn-protocol/)

**Comment by britshickhouse** [comment]
> Yes please, +1 for AdguardVPN support. There's this page about the protocol in their docs which might be useful: https://adguard-vpn.com/kb/general/adguard-vpn-protocol/

The link now redirects to [https://trusttunnel.org/](https://trusttunnel.org/) ... _"A modern, open-source VPN protocol originally developed by AdGuard VPN and now available for anyone to use, audit, and implement."_

Does this provide enough detail for gluetun to support Adguard VPN?
